precompiles/pallet-democracy/src/lib.rs

```diff
@@ -274,13 +208,17 @@ where
 
 		RuntimeHelper::<Runtime>::try_dispatch(handle, Some(origin).into(), call)?;
 
-		Ok(succeed([]))
+		Ok(())
 	}
 
-	fn second(handle: &mut impl PrecompileHandle) -> EvmResult<PrecompileOutput> {
-		// substrate arguments are u32 but Solidity one are uint256.
-		// We parse as uint32 to properly reject overflowing values.
-		read_args!(handle, {prop_index: u32, seconds_upper_bound: u32});
+	#[precompile::public("second(uint256,uint256)")]
+	fn second(
+		handle: &mut impl PrecompileHandle,
+		prop_index: SolidityConvert<U256, u32>,
+		seconds_upper_bound: SolidityConvert<U256, u32>,
+	) -> EvmResult {
+		let prop_index = prop_index.converted();
+		let seconds_upper_bound = seconds_upper_bound.converted();
```

Comment:
why are not handling here potential errors from converting u256 to u32?

Reply:
Error is handled directly by `SolidityConvert`. Here the issue is that the Solidity interface use `uint256` while it should have been `uint32`. `SolidityConvert` thus parse a `uint256` then `try_into` `u32`. If you enter the function it means it fitted into `u32` and can be unwrapped directly.

Reply:
OH I see, then if its not convertible will revert at argument reading right? Why not do the same with balances?

Reply:
We could do the same, however for balances I'd prefer it to display that it failed to parse a "balance" instead of the underlying type. We can make a wrapper for that, but that would require to write `.into()` for all balances in tests. I can do that.

Reply:
Yeah not sure if that is worth the effort. I trust your judgement

Reply:
For now I think having this function is fine. We can easily change it in a later PR :)